data/ipfs.go

```diff
@@ -50,10 +54,13 @@ type IPFSHandle struct {
 }
 
 func (i *IPFSHandle) Init(d *types.DataStore) error {
-	if i.LogLevel == "" {
-		i.LogLevel = "ERROR"
+	log = logger.Named(loggerName)
+
+	l := logger.LogLevelFor(loggerName)
+	if l == "" { // default to ERROR LogLevel
+		l = "ERROR"
 	}
```

Comment:
API-wise, it would be nicer if creating the logger also set the default log level for that component, like:

```
var log = log.Named(data.ipfs, log.Error)
```

Then it's shorter and you don't need to expose API like LogLevelFor.

Reply:
mmh here i'm trying to do the opposite, snoop into what LogLevel was set upstream for this component (`ipfs`) and pass that same level to the (underlying) ipfs

right now my approach is that:
* each component chooses their own name (ipfssync defines itself as "ipfssync")
* the caller chooses which logLevel to use for each component (so, dvotenode chooses "error" or "debug" for component "ipfssync")

but i'm thinking it'd probably be wiser and more consistent that the caller chooses BOTH the component name and loglevel. i'll rework the approach like this, i think the result will make much more sense

Reply:
in the end i kept the original approach, which is also the same implemented by `ipfs/go-log`